include/SFML/Window/Keyboard.hpp

```diff
@@ -154,11 +160,179 @@ class SFML_WINDOW_API Keyboard
 
         // Deprecated values:
 
-        Dash      = Hyphen,       //!< \deprecated Use Hyphen instead
-        BackSpace = Backspace,    //!< \deprecated Use Backspace instead
-        BackSlash = Backslash,    //!< \deprecated Use Backslash instead
-        SemiColon = Semicolon,    //!< \deprecated Use Semicolon instead
-        Return    = Enter         //!< \deprecated Use Enter instead
+        Dash      = Hyphen,    //!< \deprecated Use Hyphen instead
+        BackSpace = Backspace, //!< \deprecated Use Backspace instead
+        BackSlash = Backslash, //!< \deprecated Use Backslash instead
+        SemiColon = Semicolon, //!< \deprecated Use Semicolon instead
+        Return    = Enter,     //!< \deprecated Use Enter instead
+        Quote     = Apostrophe //!< \deprecated Use Apostrophe instead
+    };
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Scancodes
+    ///
+    /// The enumerators are bound to a physical key and do not depend on
+    /// the keyboard layout used by the operating system. Usually, the AT-101
+    /// keyboard can be used as reference for the physical position of the keys.
+    ///
+    ////////////////////////////////////////////////////////////
+    enum Scancode
```

Comment:
This is also what kim366 suggested, but didn't receive a follow up: https://github.com/SFML/SFML/pull/1235#discussion_r138327093

Until we get enum classes, I personally would stick to the prefix, so it's more "consistent" between `sf::Keyboard::Key` and `sf::Keyboard::Scancode`. But I'm also open to change it, as it would mean less refactoring for people switching to SFML 3.

Reply:
Not sure what you mean with "consistent", isn't it rather a disadvantage if both keys and scancodes have _the same_ prefix `sf::Keyboard`?

Since this feature gets rather late into SFML 2, it might be indeed nice if it's closer to SFML 3 already.

Other opinions, @binary1248 @TankOs @LaurentGomila and whoever I forgot?

Reply:
I would also personally prefer a namespace/scope. An additional problem I see with the prefix is that the user can't tell based on the name alone whether the symbol is really a key called "ScanStop" (that I don't know... commands a scanner to stop scanning or something?) or to be used as a scancode. With a namespace/scope there is no longer any ambiguity because Scan is no longer part of the name.

Reply:
Oops, this was 4 years ago, I'm can barely remember this discussion. I wonder, though, why, at even now SFML still tries to keep everything pre C++ 11

Reply:
@kim366 We will migrate to C++17 as described in this [solution design](https://github.com/SFML/SFML/wiki/SD%3A-CPP-17-Support#c-standard), however the SFML 2.* branch needs to remain backwards-compatible, so we need an intermediate solution. With that in mind, the migration is likely smoother when the solution is already as close as possible to the C++17 equivalent.

Reply:
@Bromeon should I make the `struct Scan` have a private constructor? Given that `sf::Keyboard` itself doesn't have a state, yet doesn't prevent instantiation, makes me think I shouldn't.

Or alternatively, we could change `sf::Keyboard` into a namespace with free functions and then we can create a `Scan` namespace inside. 🤔

Reply:
The more I think about this/implement it, the less certain I am whether we should really go for it...

Should it be `sf::Keyboard::Scan::Scancode` or `sf::Keyboard::Scan::Code`? Either way, you'll now have `Scan::Scancode` / `Scan::Code` as parameter/return type, which feels odd

Reply:
> @Bromeon should I make the `struct Scan` have a private constructor? Given that `sf::Keyboard` itself doesn't have a state, yet doesn't prevent instantiation, makes me think I shouldn't.

True, let's then be consistent with the rest.

> Or alternatively, we could change `sf::Keyboard` into a namespace with free functions and then we can create a `Scan` namespace inside. 🤔

That should probably have been a namespace in the first place. A big disadvantage with our current tooling is that namespaces are not prominently listed in Doxygen -- [`sf::Glsl`](https://www.sfml-dev.org/documentation/2.5.1/namespaces.php) is living a lonely existence...

And then such a change would also affect `Mouse`, `Joystick` and possibly others. I would say this needs to be discussed in a separate issue, maybe we can just do it differently for SFML 3.0. Even if little, it would still break code such as the following, essentially for no gain:
```cpp
typedef sf::Keyboard Kb;
```

Reply:
I agree that this should use something closer to `enum class` syntax. It's just "nicer" and closer to a modern C++ counterpart. I'd rather write `sf::Keyboard::ScanCode::A` compared to `sf::Keyboard::ScanA`.

Reply:
Marking this as resolved, since I have added `sf::Keyboard::Scan::X`